builtin/logical/pki/cert_template.proto

```diff
@@ -0,0 +1,211 @@
+}
+
+message IPMask {
+    bytes IPMask = 1;
+}
+
+message net {
+    message IP {
+        bytes IP = 1;
+    }
+    message IPNet {
+        net.IP IP = 1;          // network number
+	    IPMask Mask = 2;
+    }
+}
+
+message Userinfo {
+	string username      = 1;   // RFC‑2396 allows empty.
+    string password      = 2;   // Empty when not supplied.
+    bool   password_set  = 3;   // If true, then password field is meaningful.
+}
+
+message url {
+    message URL {
+        string Scheme = 1;
+        string Opaque = 2;
+        Userinfo User = 3;
+        string Host = 4;
+        string Path = 5;
+        string RawPath = 6;
+        bool OmitHost = 7;
+        bool ForceQuery = 8;
+        string RawQuery = 9;
+        string Fragment = 10;
+        string RawFragment = 11;
+    }    
+}
+
+// An OID represents an ASN.1 OBJECT IDENTIFIER.
+message OID {
+    repeated uint64 arc = 1;     // 1.2.840.113549 → [1, 2, 840, 113549]
+}
+
+message PolicyMapping {	
+	OID IssuerDomainPolicy = 1;	
+	OID SubjectDomainPolicy = 2;
+}
+
+enum PublicKeyAlgorithm {
+    UnknownPublicKeyAlgorithm = 0;
+    RSA = 1;
+    DSA = 2;
+    ECDSA = 3;
+    Ed25519 = 4;
+}
+
+enum SignatureAlgorithm {
+    UnknownSignatureAlgorithm = 0;
+	MD2WithRSA  = 1;            // Unsupported.
+	MD5WithRSA = 2;             // Only supported for signing, not verification.
+	SHA1WithRSA = 3;            // Only supported for signing, and verification of CRLs, CSRs, and OCSP responses.
+	SHA256WithRSA = 4;
+	SHA384WithRSA = 5;
+	SHA512WithRSA = 6;
+	DSAWithSHA1 = 7;            // Unsupported.
+	DSAWithSHA256 = 8;          // Unsupported.
+	ECDSAWithSHA1 = 9;          // Only supported for signing, and verification of CRLs, CSRs, and OCSP responses.
+	ECDSAWithSHA256 = 10;
+	ECDSAWithSHA384 = 11;
+	ECDSAWithSHA512 = 12;
+	SHA256WithRSAPSS = 13;
+	SHA384WithRSAPSS = 14;
+	SHA512WithRSAPSS = 15;
+	PureEd25519 = 16;
+}
+
+enum KeyUsage {
+    KeyUsageUnspecified = 0;
+    KeyUsageDigitalSignature = 1;
+    KeyUsageContentCommitment = 2;
+	KeyUsageKeyEncipherment = 3;
+	KeyUsageDataEncipherment = 4;
+	KeyUsageKeyAgreement = 5;
+	KeyUsageCertSign = 6;
+	KeyUsageCRLSign = 7;
+	KeyUsageEncipherOnly = 8;
+	KeyUsageDecipherOnly = 9;
+}
+
+enum ExtKeyUsage {
+    ExtKeyUsageAny = 0;
+	ExtKeyUsageServerAuth = 1;
+	ExtKeyUsageClientAuth = 2;
+	ExtKeyUsageCodeSigning = 3;
+	ExtKeyUsageEmailProtection = 4;
+	ExtKeyUsageIPSECEndSystem = 5;
+	ExtKeyUsageIPSECTunnel = 6;
+	ExtKeyUsageIPSECUser = 7;
+	ExtKeyUsageTimeStamping = 8;
+	ExtKeyUsageOCSPSigning = 9;
+	ExtKeyUsageMicrosoftServerGatedCrypto = 10;
+	ExtKeyUsageNetscapeServerGatedCrypto = 11;
+	ExtKeyUsageMicrosoftCommercialCodeSigning = 12;
+	ExtKeyUsageMicrosoftKernelCodeSigning = 13;
+}
+
+// Mirrors x509.Certificate
+message CertTemplate {
+    // Complete ASN.1 DER content (certificate, signature algorithm and signature).
+    bytes Raw = 1; 
+	bytes RawTBSCertificate = 2; 
+	bytes RawSubjectPublicKeyInfo  = 3; 
```

Comment:
In general, there's a few `Raw` values we'll keep but most we'll drop. 

Let's drop these three. The first two are determined by the rest of the structure and the last is determined by the provided CSR and/or generated key.